api/services/RichText.js

```diff
@@ -1,54 +1,152 @@
-var Cheerio = require('cheerio')
```

Comment:
Other than Link Preview related stuff, this file is the real heart of the changes in this branch. 

A lot of what is happening here was spread across a few places in the frontend and backend previously. 

Take a read through and let me know if you have any questions. 

---

NOTE:  I've moved here from `cheerio` to `jsdom` parsing and for now I'd like us to keep moving that way. It debatably (JSDOM has gotten a LOT better in recent years) is somewhat less performant than `cheerio` but allows for code that is more isomorphic (could be ran frontend as well) and keeps us familiar with standard DOM methods. 

We currently are using `cheerio` one other place here on `Hylo-node` and in the `Highlight...` search-related stuff on the frontend. It would be nice to convert those spots too... eventually.